File: lib-cli/tmp/server-ts.backup/src/producer/implementation/events/connected.ts
import { Producer, Identification, Filter, Context } from "./index";
import { emit } from "../../events/connected";

export function handler(
	consumer: Identification,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<void> {
	return emit(consumer, filter, context, producer);
}

File: examples/producer/rust/src/producer/implementation/emitters/connected.rs
use super::{events, identification, producer::Control, Context, EmitterError};
use fiber::server;
pub async fn emit<E: std::error::Error, C: server::Control<E> + Send + Clone>(
    identification: &mut identification::Identification,
    filter: &identification::Filter,
    context: &mut Context,
    control: &Control<E, C>,
) -> Result<(), EmitterError> {
    events::connected::emit::<E, C>(identification, filter, context, control)
        .await
        .map_err(EmitterError::Emitting)
}